### PR TITLE
Allow QAT to handle non page-aligned buffers

### PR DESCRIPTION
**Pull request by tcaputi**
This patch adds some handling to the QAT acceleration functions
that allows them to handle buffers that are not aligned with the
page cache. At the moment this never happens since callers only
happen to work with page-aligned buffers, but this code should
prevent headaches if this isn't always true in the future. This
patch also adds some cleanups to align the QAT compression code
with the encryption and checksumming code.

Signed-off-by: Weigang Li <weigang.li@intel.com>
Signed-off-by: Tom Caputi <tcaputi@datto.com>

### How Has This Been Tested?
Ran basic regression tests to ensure that QAT encryption, checksumming, and gzip all still work and are compatible with the non-accelerated implementation.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [x] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [x] My code follows the ZFS on Linux code style requirements.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] All commit messages are properly formatted and contain `Signed-off-by`.
- [ ] Change has been approved by a ZFS on Linux member.

### CODE REVIEW

**Comment by tcaputi** [comment]
@wli5 could you give this a quick look?

File: module/zfs/qat_crypt.c [hunk 1]
```diff
@@ -297,10 +295,9 @@ qat_crypt(qat_encrypt_dir_t dir, uint8_t *src_buf, uint8_t *dst_buf,
 	CpaStatus status = CPA_STATUS_SUCCESS;
 	Cpa16U i;
 	CpaInstanceHandle cy_inst_handle;
-	Cpa16U nr_bufs = (enc_len + PAGE_CACHE_SIZE - 1) / PAGE_CACHE_SIZE;
+	Cpa16U nr_bufs = (enc_len >> PAGE_SHIFT) + 1;
```
Comment (wli5) [inline module/zfs/qat_crypt.c]: nr_bufs also need consider the address is not page aligned, changes should be:
Cpa16U nr_bufs = (enc_len + ((long)src & ~PAGE_MASK)) >> PAGE_SHIFT) + 1;
Same change is also needed in qat_compress.c line 244 and 245.

File: module/zfs/qat_crypt.c [hunk 2]
```diff
@@ -439,7 +446,7 @@ qat_checksum(uint64_t cksum, uint8_t *buf, uint64_t size, zio_cksum_t *zcp)
 	CpaStatus status;
 	Cpa16U i;
 	CpaInstanceHandle cy_inst_handle;
-	Cpa16U nr_bufs = (size + PAGE_CACHE_SIZE - 1) / PAGE_CACHE_SIZE;
+	Cpa16U nr_bufs = (size >> PAGE_SHIFT) + 1;
```
Comment (wli5) [inline module/zfs/qat_crypt.c]: Cpa16U nr_bufs = (size + ((long)buf & ~PAGE_MASK)) >> PAGE_SHIFT) + 1;

**Comment by tcaputi** [comment]
@wli5 I just implemented a (slightly simplified) version of your recommendation. Let me know if it is acceptable.

**Pull request merged**